extension/core_functions/scalar/list/list_value.cpp

```diff
@@ -11,43 +11,153 @@
 	}
 };
 
-template <class T, class OP = ListValueAssign>
-static void TemplatedListValueFunction(DataChunk &args, Vector &result) {
-	idx_t list_size = args.ColumnCount();
-	ListVector::Reserve(result, args.size() * list_size);
-	auto result_data = FlatVector::GetData<list_entry_t>(result);
-	auto &list_child = ListVector::GetEntry(result);
-	auto child_data = FlatVector::GetData<T>(list_child);
-	auto &child_validity = FlatVector::Validity(list_child);
-
-	auto unified_format = args.ToUnifiedFormat();
-	for (idx_t r = 0; r < args.size(); r++) {
-		for (idx_t c = 0; c < list_size; c++) {
-			auto input_idx = unified_format[c].sel->get_index(r);
-			auto result_idx = r * list_size + c;
-			auto input_data = UnifiedVectorFormat::GetData<T>(unified_format[c]);
-			if (unified_format[c].validity.RowIsValid(input_idx)) {
-				child_data[result_idx] = OP::template Assign<T>(input_data[input_idx], list_child);
+struct ListValueListEntryAssign {
+	template <class T>
+	static T Assign(Vector &result, ListValueInfo<T> &info, idx_t input_idx, idx_t col) {
+		const auto input_data = UnifiedVectorFormat::GetData<list_entry_t>(info.unified_format[col]);
+		const auto length = input_data[input_idx].length;
+		const auto offset = (*info.col_offsets)[col] + input_data[input_idx].offset;
+		return list_entry_t(offset, length);
+	}
+};
+
+template<class T, class OP = ListValueAssign>
+static void TemplatedPopulateList(DataChunk &args, Vector &result, const optional_ptr<vector<idx_t>> col_offsets = nullptr) {
+	auto info = ListValueInfo<T>(args, result, col_offsets);
+
+	for (idx_t row = 0; row < info.row_count; row++) {
+		for (idx_t col = 0; col < info.list_size; col++) {
+			auto input_idx = info.unified_format[col].sel->get_index(row);
+			auto result_idx = row * info.list_size + col;
+			if (info.unified_format[col].validity.RowIsValid(input_idx)) {
+				info.result_child_data[result_idx] = OP::template Assign<T>(*info.result_list_child, info, input_idx, col);
 			} else {
-				child_validity.SetInvalid(result_idx);
+				info.result_child_validity->SetInvalid(result_idx);
 			}
 		}
-		result_data[r].offset = r * list_size;
-		result_data[r].length = list_size;
+	}
+
+}
+
+static void TemplatedListValueFunctionFallback(DataChunk &args, Vector &result) {
+	auto &child_type = ListType::GetChildType(result.GetType());
+	const auto result_data = FlatVector::GetData<list_entry_t>(result);
+	for (idx_t i = 0; i < args.size(); i++) {
+		result_data[i].offset = ListVector::GetListSize(result);
+		for (idx_t col_idx = 0; col_idx < args.ColumnCount(); col_idx++) {
+			auto val = args.GetValue(col_idx, i).DefaultCastAs(child_type);
+			ListVector::PushBack(result, val);
+		}
+		result_data[i].length = args.ColumnCount();
+	}
+}
+
+template<class OP = ListValueAssign>
+static void PopulateList(DataChunk &args, Vector &result, const PhysicalType child_type, const optional_ptr<vector<idx_t>> col_offsets = nullptr) {
```

Comment:
Aren't we missing the `PhysicalType::STRUCT` case here?
In case we have a STRUCT inside a STRUCT